cmake/BuildFlags.cmake

```diff
@@ -9,78 +7,84 @@ if (MSVC AND NOT ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel"))
-    /RTC1       #*Enables run-time error checking.
-  )
-  string(REPLACE ";" " " CMAKE_CXX_FLAGS_DEBUG "${CompileDebug}")
-
-  #==============#
-  # Linker flags #
-  #==============#
-
-  set(LinkCommon
-    /NOLOGO     # Suppresses the startup banner.
-    /DYNAMICBASE#Specifies whether to generate an executable image that can be randomly rebased at load time by using the address space layout randomization (ASLR) feature.
-    /MACHINE:X86 # Specifies the target platform.
-  )
-  string(REPLACE ";" " " CMAKE_EXE_LINKER_FLAGS "${LinkCommon}")
-
-  set(LinkRelease
-    /LTCG       # Specifies link-time code generation.
-    /DEBUG	    # Creates debugging information.
-    /MAP        # Creates a mapfile.
-    /INCREMENTAL:NO # Controls incremental linking.
-  )
-  string(REPLACE ";" " " CMAKE_EXE_LINKER_FLAGS_RELEASE "${LinkRelease}")
-
-  set(LinkDebug
-    /DEBUG	    # Creates debugging information.
-    /MAP        # Creates a mapfile.
-    /INCREMENTAL # Controls incremental linking.
-  )
-  string(REPLACE ";" " " CMAKE_EXE_LINKER_FLAGS_DEBUG "${LinkDebug}")
-
+target_compile_options(cse_common_interface INTERFACE
+  # MSVC common compiler flags
+  $<$<CXX_COMPILER_ID:MSVC>:
+    /DWIN32       #*
+    # /D_WINDOWS  #*
+    /D_CONSOLE    #
+    /DINCNE       # CSE-specific
+    /W3           #*Warning level
+    # /WX           # Turn all warnings into errors
+    /GR           #*Enable Run-Time Type Information TODO: keep?
+    /EHsc         #*Specifies the model of exception handling (sc options).
+    /nologo       # Suppresses display of sign-on banner.
+    /fp:precise   #+Specifies floating-point behavior.
+    /fp:except-   # Specifies floating-point behavior.
+    /arch:IA32    # Specifies the architecture for code generation (no special instructions).
+    # MSVC Release Flags
+    $<$<CONFIG:Release>:
+      # /DNDEBUG  #* TODO: Add back?
+      # /MD       #*Creates a multithreaded DLL using MSVCRT.lib.
+      /MT         # Creates a multithreaded executable file using LIBCMT.lib.
+      /GF         # Enables string pooling.
+      /GL         # Enables whole program optimization.
+      /Gy         # Enables function-level linking.
+      # /O2       #*Creates fast code.
+      # /Ob2      #*Controls inline expansion (level 2).
+      /Oi         # Generates intrinsic functions.
+      /Ot         # Favors fast code.
+      /Ox         # Uses maximum optimization (/Ob2gity /Gs).
+      /Oy         # Omits frame pointer (x86 only).
+      /Gd         # Uses the __cdecl calling convention (x86 only).
+      /Zi         #*Generates complete debugging information.
+      /GS-        # Buffers security check.
+      /MP         # Compiles multiple source files by using multiple processes.
+    >
+    # # MSVC DEBUG Flags
+    $<$<CONFIG:Debug>:
+      /D_DEBUG    #*
+      # /MDd      #*Creates a debug multithreaded DLL using MSVCRT.lib.
+      /MTd        # Creates a debug multithreaded executable file using LIBCMTD.lib.
```

Comment:
I know this is pulled from the existing comments, but technically the /MT and /MTd flags don't "create" a multithreaded executable, they just use the multithreaded version of MS runtime libraries.